http.go

```diff
@@ -213,6 +213,24 @@ func (c httpClient) KeyReadAll(ctx context.Context) ([]JWK, error) {
 	}
 	return jwks, nil
 }
+func (c httpClient) KeyReplaceAll(ctx context.Context, given []JWK) error {
+	err := c.given.KeyReplaceAll(ctx, given)
+	if err != nil {
+		return fmt.Errorf("failed to delete all keys from given storage due to error: %w", err)
+	}
+	var returnErr error
+	for _, store := range c.httpURLs {
+		err = store.KeyReplaceAll(ctx, make([]JWK, 0))
+		if err != nil {
+			if returnErr == nil {
+				returnErr = fmt.Errorf("failed to delete all keys: %w", err)
+			} else {
+				returnErr = errors.Join(returnErr, fmt.Errorf("failed to delete all keys: %w", err))
+			}
```

Comment:
nit:
`errors.Join()` handles it when passing nil for errors. You can omit the nested `if`

https://go.dev/play/p/Z2XTOsFOYPM